squad/core/queries.py

```diff
@@ -195,3 +196,30 @@ def get_dynamic_summary(project, environments, metrics, date_start, date_end):
         entry[environment.slug] = sorted(envdata, key=(lambda e: e[0]))
 
     return entry
+
+
+def test_confidence(test):
+    status_priority = ['pass', 'fail', 'xfail', 'skip']
+
+    def most_common(lst):
+        data = Counter(lst)
+        max_count = max(data.values())
+        return {value: count for value, count in data.items() if count == max_count}
+
+    duplicates = models.Test.objects.filter(suite=test.suite, metadata=test.metadata, test_run=test.test_run, environment=test.environment, build=test.build)
```

Comment:
By default all queries to Test will have `order_by('metadata__name')`. I think you can avoid that by `order_by()`